plugins/tedge_apt_plugin/src/main.rs

```diff
@@ -238,33 +276,34 @@ fn run_cmd(cmd: &str, args: &str) -> Result<ExitStatus, InternalError> {
 }
 
 fn get_name_and_version(line: &str) -> (&str, &str) {
-    let vec: Vec<&str> = line.split(' ').collect();
-    // The first element should always exist and has slash, like tedge/tedge-main
-    let name = vec
-        .first()
-        .unwrap_or(&"unknown name")
-        .split_once('/')
-        .unwrap_or(("unknown name", "others"))
-        .0;
-    let version = if line.contains("upgradable from:") {
-        // e.g. tedge/tedge-main 0.8.1-127-gb7d7d599 arm64 [upgradable from: 0.8.1]
-        vec.get(5)
-            .unwrap_or(&"unknown version")
-            .trim_end_matches(']')
-    } else {
-        // e.g. tedge_agent/now 0.8.1 arm64 [installed,local]
-        vec.get(1).unwrap_or(&"unknown version")
-    };
+    let vec: Vec<&str> = line.split('\t').collect();
+
+    let name = vec.first().unwrap_or(&"unknown name");
+    let version = vec.get(1).unwrap_or(&"unknown version");
     (name, version)
 }
 
+fn get_config() -> Option<TedgeConfig> {
+    let config_dir = TEdgeConfigLocation::default();
+
+    match fs::read_to_string(config_dir.tedge_config_file_path()) {
```

Comment:
Wondering why we chose to parse the tedge.toml file directly instead of using the `TEdgeConfig` APIs. To avoid an extra dependency for the plugin?

Reply:
Got the answer from @didier-wenzek 's previous review comments.